CRM/Contribute/BAO/Contribution.php

```diff
@@ -207,12 +207,11 @@ public static function add(&$params, $ids = []) {
       self::recordFinancialAccounts($params);
     }
 
-    if (self::isUpdateToRecurringContribution($params)) {
-      CRM_Contribute_BAO_ContributionRecur::updateOnNewPayment(
-        (!empty($params['contribution_recur_id']) ? $params['contribution_recur_id'] : $params['prevContribution']->contribution_recur_id),
-        $contributionStatus,
-        CRM_Utils_Array::value('receive_date', $params)
-      );
+    $recurID = self::isUpdateToRecurringContribution($params);
+    if ($recurID) {
+      $paymentProcessorID = (int) civicrm_api3('ContributionRecur', 'getvalue', ['id' => $recurID, 'return' => 'payment_processor_id']);
```

Comment:
Not all recurrings have processor ids - ie imported ones - this will break them

Reply:
@eileenmcnaughton The cast to int converts NULL to 0 which maps to the manual processor so any without a processor will use that automatically.

Reply:
But will the api call through an exception or return ''?